### PR TITLE
Prepare RecordFileParser for file init/complete logic.

### LINKED ISSUE #566: [parser refactor] Split PostgresWritingRecordParsedItemHandler out of RecordFileLogger
Move all connection and PreparedStatement logic out of RecordFileLogger
- `PostgresWritingRecordParsedItemHandler` will
  - implement `RecordParsedItemHandler`
  - own connection, setup PreparedStatements.
  - handle batching logic
  - Do `f_file_create` (in `initFile`) and `f_file_complete` (in `completeFile`)
- Keep all Transaction/TransactionRecord parsing logic in `RecordFileLogger` (to be renamed separately to keep PRs small)

### PR DESCRIPTION
**Pull request by apeksharma**
**Detailed description**:

- Right now record file init/complete logic is in RecordFileLogger. In new design, we want to move
  it to RecordFileParser.
- This change (part 1 of 2) refactors RFP so that actual logic move (followup, part 2 of 2) can be neat.
- New functions - initFile, closeFileAndCommit, rollback - are where the logic will be moved to.
- Make loadRecordFile filesystem agnostic.
- Although a lot of code being touched here can be improved in many other ways, keeping this PR refactor only for easy review

Signed-off-by: Apekshit Sharma <apekshit.sharma@hedera.com>

**Which issue(s) this PR fixes**:
Partially fixes #566 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java [hunk 1]
```diff
@@ -120,37 +123,45 @@ public static String readPrevFileHash(String fileName) {
         return null;
     }
 
+    private RecordFileLogger.INIT_RESULT initFile(String filename) {
```
Comment (apeksharma) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java]: these 3 functions look funny right now. In followup, the logic from RFLogger will move to these functions.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java [hunk 2]
```diff
@@ -120,37 +123,45 @@ public static String readPrevFileHash(String fileName) {
...
-     * @param thisFileHash     the hash of this file
+     * @param fileName             the name of record file to read
+     * @param expectedPrevFileHash the hash of the previous record file in the series
+     * @param thisFileHash         the hash of this file
      * @return return boolean indicating method success
      * @throws Exception
      */
-    private boolean loadRecordFile(String fileName, String previousFileHash, String thisFileHash) throws Exception {
-
-        File file = new File(fileName);
```
Comment (apeksharma) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java]: moved filesystem dependent code to one level up.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java [hunk 3]
```diff
@@ -120,37 +123,45 @@ public static String readPrevFileHash(String fileName) {
...
         long counter = 0;
-        byte[] readFileHash = new byte[48];
-        RecordFileLogger.INIT_RESULT initFileResult = RecordFileLogger.initFile(fileName);
         Stopwatch stopwatch = Stopwatch.createStarted();
         Integer recordFileVersion = 0;
         Boolean success = false;
 
-        if (initFileResult == RecordFileLogger.INIT_RESULT.OK) {
-            try (DataInputStream dis = new DataInputStream(new FileInputStream(file))) {
+            try (DataInputStream dis = new DataInputStream(inputStream)) {
```
Comment (apeksharma) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java]: re-aligning is throwing the differ off, and results in big lumps of red and green.
Keeping it un-aligned explicitly for easy review.
On approval, will push a commit to shift code left.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java [hunk 4]
```diff
@@ -120,37 +123,45 @@ public static String readPrevFileHash(String fileName) {
...
+    }
+
     /**
      * Given a service record name, read and parse and return as a list of service record pair
      *
-     * @param fileName         the name of record file to read
-     * @param previousFileHash the hash of the previous record file in the series
-     * @param thisFileHash     the hash of this file
+     * @param fileName             the name of record file to read
+     * @param expectedPrevFileHash the hash of the previous record file in the series
```
Comment (Nana-EC) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java]: nit: missing inputStream

Comment (apeksharma) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java]: done.

**Comment by apeksharma** [comment]
Thanks for the review @Nana-EC

**Pull request merged**